cpp-client/deephaven/dhcore/include/public/deephaven/dhcore/utility/utility.h

```diff
@@ -238,4 +243,43 @@ inline void trueOrThrow(const DebugInfo &debugInfo, bool value) {
   }
   internal::trueOrThrowHelper(debugInfo);
 }
+
+[[nodiscard]] std::string
+EpochMillisToStr(std::int64_t epochMillis);
+
+[[nodiscard]] inline std::int64_t
+TimePointToEpochMillis(
+    const std::chrono::time_point<std::chrono::system_clock> timePoint) {
+  using namespace std::chrono;
+  const milliseconds ms = duration_cast<milliseconds>(timePoint.time_since_epoch());
+  return ms.count();
+}
+
+[[nodiscard]] inline std::string
+TimePointToStr(
+    const std::chrono::time_point<std::chrono::system_clock> timePoint) {
+  return EpochMillisToStr(TimePointToEpochMillis(timePoint));
+}
+
+[[nodiscard]] std::string Demangle(const char* name);
+
+template <class T> [[nodiscard]] std::string
+TypeName(const T& t) {
+  return Demangle(typeid(t).name());
+}
+
+[[nodiscard]] inline std::string
+ObjectId(const std::string &classShortName, void* thisPtr) {
+  std::string id(classShortName + "[0x0000000000000000]");
+  snprintf(&id[classShortName.size() + 3], 16, "%p", thisPtr);
+  return id;
+}
+
+inline std::string
+ThreadIdToString(const std::thread::id tid) {
+  std::stringstream ss;
+  ss << tid;
+  return ss.str();
+}
```

Comment:
Here I would again use `SimpleOstringstream` and `return std::move(ss.str())`

or even the kosakian

```
std::string ThreadIdToString(const std::thread::id tid) {
  return stringf("%o", tid);
}
```